src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -3514,5 +3514,56 @@ public void M(ref Test obj)
                 //         this = ref obj;
                 Diagnostic(ErrorCode.ERR_RefLocalOrParamExpected, "this").WithLocation(8, 9));
         }
+
+        [Fact]
+        [WorkItem(29927, "https://github.com/dotnet/roslyn/issues/29927")]
+        public void CoalesceSpanReturn()
+        {
+            CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class C
+{
+    Span<byte> M()
+    {       
+        return null ?? new Span<byte>();
+    }
+}", options: TestOptions.ReleaseDll).VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(29927, "https://github.com/dotnet/roslyn/issues/29927")]
+        public void CoalesceAssignSpanReturn()
+        {
+            CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class C
+{
+    Span<byte> M()
+    {       
+        var x = null ?? new Span<byte>();
+        return x;
+    }
+}", options: TestOptions.ReleaseDll).VerifyDiagnostics();
+        }
+
+        [Fact]
+        [WorkItem(29927, "https://github.com/dotnet/roslyn/issues/29927")]
+        public void CoalesceRefSpanReturn()
+        {
+            CreateCompilationWithMscorlibAndSpan(@"
+using System;
+class C
+{
+    Span<byte> M()
+    {       
+        Span<byte> x = stackalloc byte[10];
+        return null ?? x;
+    }
+}", options: TestOptions.ReleaseDll).VerifyDiagnostics(
+                // (8,24): error CS8352: Cannot use local 'x' in this context because it may expose referenced variables outside of their declaration scope
+                //         return null ?? x;
+                Diagnostic(ErrorCode.ERR_EscapeLocal, "x").WithArguments("x").WithLocation(8, 24)
+                );
+        }
```

Comment:
@333fred we should add an issue to check this behavior for `??=` as well.

Reply:
@chsienki do you want to open a separate issue, or just add checks in this PR? `??=` should make it into dev16.0.x later today, and from there should be in master within a day or two.

---
In reply to: [218519137](https://github.com/dotnet/roslyn/pull/29985#discussion_r218519137) [](ancestors = 218519137)

Reply:
Hmm, it looks like we're indirectly testing this https://github.com/dotnet/roslyn/blob/01fcdf971014d9ce2b36f3edcf7c840b1c7728fc/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCoalescingAssignmentTests.cs#L2081

But we should probably add an explicit test for non-nullable reference types. I'll open an issue rather than complicating this PR.